students/ArchWu/lesson02/assignment/src/charges_calc.py

```diff
@@ -0,0 +1,91 @@
+'''
+Returns total price paid for individual rentals
+'''
+import argparse
+import json
+import datetime
+import math
+import logging
+
+LOG_FORMAT = "%(asctime)s %(filename)s:%(lineno)-3d %(levelname)s %(message)s"
+LOG_FILE = datetime.datetime.now().strftime("%Y-%m-%d")+'.log'
+FILE_HANDLER = logging.FileHandler(LOG_FILE)
+FORMATTER = logging.Formatter(LOG_FORMAT)
+#LEVEL = {0: logging.NOTSET, 1:logging.ERROR, 2:logging.WARNING, 3:logging.INFO}
+CONSOLE_HANDLER = logging.StreamHandler()
+CONSOLE_HANDLER.setFormatter(FORMATTER)
+LOGGER = logging.getLogger()
+LOGGER.addHandler(FILE_HANDLER)
+LOGGER.addHandler(CONSOLE_HANDLER)
+
+def parse_cmd_arguments():
+    """Command Parser"""
+    parser = argparse.ArgumentParser(description='Process some integers.')
+    parser.add_argument('-i', '--input', help='input JSON file', required=True)
+    parser.add_argument('-o', '--output', help='ouput JSON file', required=True)
+    parser.add_argument('-d', '--debug', help='debug option',
+                        required=True, type=int)
+    return parser.parse_args()
+
+
+def load_rentals_file(filename):
+    """File loader"""
+    file = open(filename, 'r')
+    try:
+        data = json.load(file)
+        logging.info("Data loading Hajime")
+        return data
+    except json.decoder.JSONDecodeError:
```

Comment:
Good catch! Since you have caught an error, it would be a good idea to log it


Also, you could run into another potential error. What if the file doesn't exist?

Reply:
Since you have imported json, you can just specify this as except JSONDecodeError